views/admin/index.php

```diff
@@ -125,7 +125,7 @@
                     }
                 },
                 'switch' => function ($url, $model) {
-                    if($model->id != Yii::$app->user->id && Yii::$app->getModule('user')->enableImpersonateUser) {
+                    if($model->isAdmin && $model->id != Yii::$app->user->id && Yii::$app->getModule('user')->enableImpersonateUser) {
```

Comment:
Hello, why actually adding this $model->isAdmin check here?
Usually $model will be the user cycled in the grid row: so $model->isAdmin will be true only for admin users.
Result: the button will not be displayed if the gridrow contains a normal user (read: the button "impersonate" will not be shown).

Reply:
@d4rkstar 
> Result: the button will not be displayed if the gridrow contains a normal user (read: the button "impersonate" will not be shown).
this is correct behavior. Impersonate available only on "admin". See origin thread discussion

Reply:
I insist: "The impersonate should be available to admin with access to all other users."

So: $model will be the user cycled in the grid row: so $model->isAdmin will be true only for admin users.

This will break the behavior you said.

Reply:
i.e. 
```php
$model->isAdmin 
```
should be replace to
```php
\Yii::$app->user->identity->isAdmin 
```
?

Reply:
isAdmin 
after 
https://github.com/dektrium/yii2-user/commit/b7180b29f3f0673b4ee5f19f5974cc2e95b93cd6

Reply:
before this commit $someuser->isAdmin will be use wrong behavior (correct for this case)

Thanks.

Reply:
So... but in this fork I can see old code
https://github.com/dabilite/yii2-user/blob/4805971bb484eeb387b36816c37e4a1ad2cc7fef/models/User.php#L125
```php
                \Yii::$app->user->can($this->module->adminPermission) : false)
```

@d4rkstar you sure use this fork?